### PR TITLE
[Doc] Update contribution guidelines and PR template

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

Update contribution guidelines and pull request template.

# All Promptflow Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -3,9 +3,10 @@
 Please add an informative description that covers that changes made by the pull request and link all relevant issues.
 
 # All Promptflow Contribution checklist:
-- [ ] **The pull request does not introduce [breaking changes]**
+- [ ] **The pull request does not introduce [breaking changes].**
 - [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
 - [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
+- [ ] **An issue is linked to the pull request.**
```
Comment (wangchao1230) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: Create an issue and linked to the pull request to get dedicated review from promptflow team. learn more: link to contributing doc.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -89,8 +87,10 @@ We use and recommend the following workflow:
 6. Run the relevant scripts in [the section below](https://github.com/microsoft/promptflow/blob/main/CONTRIBUTING.md#dev-scripts) to ensure that your build is clean and all tests are passing.
 7. Create a PR against the repository's **main** branch.
    - State in the description what issue or improvement your change is addressing.
+   - Link the PR to the issue in step 1.
    - Verify that all the Continuous Integration checks are passing.
 8. Wait for feedback or approval of your changes from the code maintainers.
+   - If there is no response for a few days, you can create an issue to raise awareness; we will add issue assignee who will contact you and loop related internal reviewer to follow up.
```
[the remark this answers is not in the document]
Comment (zhengfeiwang) [reply CONTRIBUTING.md:93]: offline synced, keep this step but no need to define SLA for now.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (zhengfeiwang) [reply .github/PULL_REQUEST_TEMPLATE.md]: updated

**Review by wangchao1230 — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -89,8 +87,12 @@ We use and recommend the following workflow:
 6. Run the relevant scripts in [the section below](https://github.com/microsoft/promptflow/blob/main/CONTRIBUTING.md#dev-scripts) to ensure that your build is clean and all tests are passing.
 7. Create a PR against the repository's **main** branch.
    - State in the description what issue or improvement your change is addressing.
+   - Link the PR to the issue in step 1.
    - Verify that all the Continuous Integration checks are passing.
 8. Wait for feedback or approval of your changes from the code maintainers.
+   - If there is no response for a few days, you can create an issue to raise awareness.
```
Comment (eniac871) [inline CONTRIBUTING.md:93]: We already mentioned create an issue at line 75, then what's the create an issue here? same one or a new one?

Comment (zhengfeiwang) [reply CONTRIBUTING.md:93]: A new one issue here.

For example, a contributor created an issue, discussed with us and assigned to himself, then we might lose our attention to this issue (and its related PR); for this case, he can raise the ad-hoc issue for our attention.

Comment (zhengfeiwang) [reply CONTRIBUTING.md:93]: Also, we explain our daily triage process here, so contributors can somehow know how to raise their awareness, no matter what they want to catch our eyes.

Comment (zhengfeiwang) [reply CONTRIBUTING.md]: offline synced, updated to highlight the new issue

**Review by eniac871 — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Pull request merged**